### ISSUE TITLE
Android error with RTCStream.release()

### ISSUE THREAD

**Issue by ndarilek** [oney/react-native-webrtc#40]
Just upgraded from 0.3 to 0.4 this morning and now some of my `.release()` calls are failing. Here is a sample:

```
  hangUp() {
    const pc = this.state.peerConnection
    if(this.state.session) {
      this.state.session.destroy()
      this.setState({session: null})
    }
    if(this.state.localStream) {
      if(pc)
        pc.removeStream(this.state.localStream)
      this.state.localStream.release()
      this.setState({localStream: null})
    }
    if(this.state.remoteStream) {
      if(pc)
        pc.removeStream(this.state.remoteStream)
      //this.state.remoteStream.release()
      this.setState({remoteStream: null})
    }
    if(this.state.peerConnection) {
      this.state.peerConnection.close()
      this.setState({peerConnection: null})
    }
    this.setState({localStreamURL: null, remoteStreamURL: null})
  }
```

If the above line is uncommented, I get the error in these screenshots. Interestingly, it doesn't appear in the adb logcat output at all, nor is the error reported to have any origins in my code:

https://dl.dropboxusercontent.com/u/147071/Screenshots/WebRTCCrash/Screenshot_20160407-102642.png

https://dl.dropboxusercontent.com/u/147071/Screenshots/WebRTCCrash/Screenshot_20160407-102647.png

This worked yesterday on 0.3.X, so I feel like something in 0.4.0 must have regressed--that or I was releasing my streams wrong in 0.3 and 0.4 is catching that but not being helpful. :)

Thanks.

**Comment by oney** [comment]
```
    if(this.state.remoteStream) {
      if(pc)
        pc.removeStream(this.state.remoteStream)
      //this.state.remoteStream.release()                        <---- this line
      this.setState({remoteStream: null})
    }
```
Did your app crash on commenting this line?

**Comment by ndarilek** [comment]
No, on *uncommenting* it.

**Comment by oney** [comment]
You don't need to release remoteStream, remoteStream will be released after peerConnection is closed.

**Comment by oney** [comment]
You should only release localStream(generated by getUserMedia) because the module doesn't know if you will use it later. The remote stream should be released after its peerConnection is closed.

**Comment by ndarilek** [comment]
Got it. Is that true for the local stream as well?

Even so, is there any way the native code can detect the double release 
and just make it a no-op? I can remove the release call, I just like 
being super cautious in my coding, because you never know when something 
may not release the RTCPeerConnection, the underlying code changes and 
the streams aren't released, etc. :)

Thanks.

**Comment by oney** [comment]
Yeah, making double release a no-op should be good. I will do that. Thanks.

**Comment by ndarilek** [comment]
No, thank *you*. :) I'll go ahead and remove the line, glad to know my 
test code isn't leaking streams.

**Comment by oney** [comment]
Update in this commit https://github.com/oney/react-native-webrtc/commit/2def8452589a8ff4abab8ab2558ad0fce843927e
Not in npm because it's fucking hard to publish new version to npm recently.

**Comment by zxcpoiu** [comment]
haha, because of this I guess.

[How one developer just broke Node, Babel and thousands of projects in 11 lines of JavaScript](http://m.theregister.co.uk/2016/03/23/npm_left_pad_chaos/)

**Comment by zxcpoiu** [comment]
fixed by 2def8452589a8ff4abab8ab2558ad0fce843927e and 48c8dd72f2d7b278ba08ae353ad10ecbb26e06ff
